quafu/qfasm/qfasm_lexer.py

```diff
@@ -0,0 +1,197 @@
+# Qfasm is modified OPENQASM 2.0. Currently it is mainly used to 
+# transfer circuit data to backends. Further it will 
+# support more instructions (classical or quantum) to enable 
+# interaction with quantum hardware
+
+import os
+import ply.lex as lex
+from .qfasm_utils import Id
+from .exceptions import LexerError
+
+class QfasmLexer(object):
+
+    def __init__(self, filename:str = None):
+        self.build(filename)
+        self.file_lexer_stack = []
+
+    def build(self, filename:str = None):
+        self.lexer = lex.lex(module=self)
+        self.lexer.filename = filename
+        self.lexer.lineno = 1
+        if filename:
+            with open(filename,"r") as ifile:
+                self.data = ifile.read()
+            self.lexer.input(self.data)
+    
+    def push_lexer(self, filename):
+        self.file_lexer_stack.append(self.lexer)
+        self.build(filename=filename)
+    
+    def pop_lexer(self):
+        self.lexer = self.file_lexer_stack.pop()
+
+    def input(self, data):
+        self.data = data
+        self.lexer.input(data)
+        # read qelib1.inc
```

Comment:
This file needs to be installed through `pip install .` You can refer to this link to install non-py files: https://setuptools.pypa.io/en/latest/userguide/datafiles.html

And this is the root cause of ut failure: https://github.com/ScQ-Cloud/pyquafu/actions/runs/6186274257/job/16793521132